packages/databricks-sdk-js/src/apierr.ts

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+const transientErrorStringMatches = [
+    "com.databricks.backend.manager.util.UnknownWorkerEnvironmentException",
+    "does not have any associated worker environments",
+    "There is no worker environment with id",
+    "Unknown worker environment",
+    "ClusterNotReadyException",
+    "connection reset by peer",
+    "TLS handshake timeout",
+    "connection refused",
+    "Unexpected error",
+    "i/o timeout",
+];
+
+export class HttpError extends Error {
+    constructor(readonly message: string, readonly code: number) {
+        super(message);
+    }
+}
+
+export class ApiError extends Error {
+    constructor(
+        public readonly message: string,
+        public readonly errorCode: string,
+        public readonly statusCode: number,
+        public readonly response: any
+    ) {
+        super(message);
+    }
+
+    isRetryable(): boolean {
+        if (this.statusCode === 429) {
+            // Repos API returns 429 when the user has too many repos. Don't retry in this case.
+            if (
+                this.message.includes(
+                    "Please delete repos before creating new repos"
+                )
+            ) {
+                return false;
+            }
+            return true;
+        }
+        if (this.statusCode >= 400) {
+            if (this.isTransientError(this.message)) {
+                return true;
```

Comment:
I like this. Let's also do something similar for other retriable errors in sdk. Maybe we can have a common "SDKError". Adding it to this ticket https://databricks.atlassian.net/browse/DECO-709